test/rocprim/test_utils_hipgraphs.hpp

```diff
@@ -27,65 +28,74 @@
 
 // Helper functions for testing with hipGraph stream capture.
 // Note: graphs will not work on the default stream.
-namespace test_utils
-{
     
-inline hipGraph_t createGraphHelper(hipStream_t& stream, const bool beginCapture=true)
-{
-    // Create a new graph
-    hipGraph_t graph;
-    HIP_CHECK(hipGraphCreate(&graph, 0));
-
-    // Optionally begin stream capture
-    if (beginCapture)
-        HIP_CHECK(hipStreamBeginCapture(stream, hipStreamCaptureModeGlobal));
-
-    return graph;
-}
-
-inline void cleanupGraphHelper(hipGraph_t& graph, hipGraphExec_t& instance)
-{
-    HIP_CHECK(hipGraphDestroy(graph));
-    HIP_CHECK(hipGraphExecDestroy(instance));
-}
-
-inline void resetGraphHelper(hipGraph_t& graph, hipGraphExec_t& instance, hipStream_t& stream, const bool beginCapture=true)
-{
-    // Destroy the old graph and instance
-    cleanupGraphHelper(graph, instance);
-
-    // Create a new graph and optionally begin capture
-    graph = createGraphHelper(stream, beginCapture);
-}
-
-inline hipGraphExec_t endCaptureGraphHelper(hipGraph_t& graph, hipStream_t& stream, const bool launchGraph=false, const bool sync=false)
-{
-    // End the capture
-    HIP_CHECK(hipStreamEndCapture(stream, &graph));
-
-    // Instantiate the graph
-    hipGraphExec_t instance;
-    HIP_CHECK(hipGraphInstantiate(&instance, graph, nullptr, nullptr, 0));
-
-    // Optionally launch the graph
-    if (launchGraph)
-        HIP_CHECK(hipGraphLaunch(instance, stream));
-
-    // Optionally synchronize the stream when we're done
-    if (sync)
-        HIP_CHECK(hipStreamSynchronize(stream));
-
-    return instance;
-}
-
-inline void launchGraphHelper(hipGraphExec_t& instance, hipStream_t& stream, const bool sync=false)
+namespace test_utils
 {
-    HIP_CHECK(hipGraphLaunch(instance, stream));
-
-    // Optionally sync after the launch
-    if (sync)
-        HIP_CHECK(hipStreamSynchronize(stream));
-}
+    class GraphHelper{
+        private:
+            hipGraph_t graph;
+            hipGraphExec_t graph_instance;
+        public:
+
+            inline void startStreamCapture(hipStream_t & stream){
+                HIP_CHECK(hipStreamBeginCapture(stream, hipStreamCaptureModeGlobal));
+            }
+
+            inline void endStreamCapture(hipStream_t & stream){
+                HIP_CHECK(hipStreamEndCapture(stream, &graph));
+            }
+            
+            inline void createGraph(){
+                HIP_CHECK(hipGraphInstantiate(&graph_instance, graph, nullptr, nullptr, 0));
+            }
+
+            inline void launchGraph(hipStream_t & stream, const bool sync=false){
```

Comment:
I think this function is the same as launchGraphHelper, below. We could probably remove one of them.

Reply:
Done :D